yt_dlp/extractor/crunchyroll.py

```diff
@@ -1,28 +1,37 @@
 import base64
-import urllib.parse
+from urllib.error import HTTPError
 
 from .common import InfoExtractor
 from ..utils import (
     ExtractorError,
     float_or_none,
     format_field,
+    int_or_none,
     join_nonempty,
+    parse_age_limit,
+    parse_count,
     parse_iso8601,
     qualities,
+    remove_start,
+    time_seconds,
     traverse_obj,
-    try_get,
+    url_or_none,
+    urlencode_postdata,
 )
 
 
 class CrunchyrollBaseIE(InfoExtractor):
-    _LOGIN_URL = 'https://www.crunchyroll.com/welcome/login'
+    _BASE_URL = 'https://www.crunchyroll.com'
     _API_BASE = 'https://api.crunchyroll.com'
     _NETRC_MACHINE = 'crunchyroll'
-    params = None
+    _AUTH_HEADERS = None
+    _API_ENDPOINT = None
+    _BASIC_AUTH = None
+    _QUERY = {}
```

Comment:
Usually caps are used for constants, no?

Reply:
- `_API_ENDPOINT` is a "constant" to be overridden by subclasses
- `_BASIC_AUTH` is a "get once and dont change" (was even debated over if we should hard code)
- `_QUERY` is append only (we just gradually build a lookup)

- `_AUTH_HEADERS` might change a bit more often, once every auth expiry. This could potentially be re... cased?

Reply:
Looking at other IEs, the case used for tokens (set once don't change) seem to be inconsistent. You can keep this as is